### ISSUE TITLE
Work with NPC team to incorporate quests into NPC struct

### ISSUE THREAD

**Issue by elizabethli31** [uchicago-cs/chiventure#1263]
Certain quests have features to meet or kill NPCs, but quests are currently not included in the NPC struct. We'd like to include quests in ways such as dialogue. This would also need to take into account prereqs to see if the player can even battle the NPC

**Comment by elizabethli31** [comment]
Started conversation with NPC team about incorporating quest into their struct. We are asking them about the possibilities of adding quests into the dialogue.

**Comment by elizabethli31** [comment]
The NPC team was working on dialogue and actions during spring 3. We will be working with them to see where quests can best be implemented. In this sprint, we came up with some design ideas and will need to run them by the NPC team. I added everything to the wiki page "Incorporating Quests into NPC"

**Comment by elizabethli31** [comment]
I made some changed to the wiki page from this original design. Originally, I had intended for a list of quests and tasks to be placed in the NPC struct. There would be two options for this method: 1) NPC dialogue triggers the initialization of certain quests or tasks given key words, 2) The NPC quest and task structs are already initialized and the dialogue is based off that. With the second option, the NPC would need to match its quest_id or task_id with the player's and see if the player is one they would interact with.
